keras_nlp/layers/transformer_decoder.py

```diff
@@ -212,35 +212,57 @@ def call(
             encoder_sequence: a Tensor. The encoder input sequence. For decoder
-                only models (like GPT2), this should be left None. Once the
+                only models (like GPT2), this should be left `None`. Once the
                 model is called once without an encoder_sequence, you cannot
                 call it again with encoder_sequence.
             decoder_padding_mask: a boolean Tensor, the padding mask of decoder
-                sequence, must of shape [batch_size, decoder_sequence_length].
+                sequence, must be of shape
+                `[batch_size, decoder_sequence_length]`.
             decoder_attention_mask: a boolean Tensor. Customized decoder
-                sequence mask, must of shape
-                [batch_size, decoder_sequence_length, decoder_sequence_length].
+                sequence mask, must be of shape
+                `[batch_size, decoder_sequence_length, decoder_sequence_length]`.
             encoder_padding_mask: a boolean Tensor, the padding mask of encoder
-                sequence, must of shape [batch_size, encoder_sequence_length].
+                sequence, must of shape `[batch_size, encoder_sequence_length]`.
             encoder_attention_mask: a boolean Tensor. Customized encoder
                 sequence mask, must of shape
-                [batch_size, encoder_sequence_length, encoder_sequence_length].
-            cache: a dense float Tensor. The cache of key/value of leading
-                tokens. `cache` is of shape [B, 2, max_seq_len, num_heads,
-                key_dims].
-            cache_index: a int or int Tensor, the index of the current token
-                being processed.
+                `[batch_size, encoder_sequence_length, encoder_sequence_length]`.
+            self_attention_cache: a dense float Tensor. The cache of key/value
+                of leading tokens in the self-attention layer. Of shape
+                `[batch_size, 2, max_seq_len, num_heads, key_dims]`.
+            self_attention_cache_update_index: an int or int Tensor, the index
+                of the current token (in `decoder_sequence`) being processed.
+                Both `self_attention_cache` and
+                `self_attention_cache_update_index` have to be non-`None` for
+                the cache to be updated.
+            cross_attention_cache: a dense float Tensor. The cache of
+                key/value pairs of encoder outputs. `cache` is of shape
+                `[batch_size, 2, S, num_heads, key_dims]`.
+            cross_attention_cache_update_index: an int or int Tensor, the index
+                of the current token (in `encoder_sequence`) being processed.
+                Since we don't actually process `encoder_sequence`
+                token-by-token, `cross_attention_cache_update_index` is given
+                a value of 0 so that the cache can be updated for all encoder
+                tokens in one go. Both `cross_attention_cache` and
```

Comment:
let's remove this last sentence and add an error

Reply:
Why error? The cache can be non-None, and the update index can be None when we want the cache to just pass through, right?